### PR TITLE
Fixes #73: Updated README and CONTRIBUTING

### LINKED ISSUE #73: Updating/Adding to readme file
A README file is a text file that contains information for the user about the software program. README files often contain instructions and additional help, and details about patches or updates. Our README file is only a couple of lines, and I will be adding information on how our system works.

### PR DESCRIPTION
**Pull request by cindyledev**
Updated README.md and CONTRIBUTING.md with

- License Section
- Git as a prereq for contributing
- Solution to a common Redis error on Linux

Ref #73 . Do not close this issue as @MusaBajwa is also assigned to it.

### CODE REVIEW

**Comment by humphd** [comment]
This has too many unrelated commits in here, I assume from a merge?  Please rebase PRs instead of merging master into them.

**Comment by cindyledev** [comment]
I got rid of the unrelated commits

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -40,6 +40,26 @@ An easier solution would be to use Docker.
...
 1. IF eslint detect some issues run `npm run eslint-fix` before manually fixing the issue (Will save you time :smile:) and then run `npm test` again.
 
+### Common Errors
+**Linux**
+
+If you encounter this error when running redis-server:
+```
+# Could not create server TCP listening socket *:6379: bind: Address already in use
+```
+1. Run `sudo systemctl status redis` to check if the service is running
```
Comment (miggs125) [inline docs/CONTRIBUTING.md]: This is unnecessary. Regular users should not have to access ```systemctl``` in this scenario. We are creating a docker container so errors addressed in this section should pertain to those that could be had from running redis in a docker container. Please read #135 for more information.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -40,6 +40,26 @@ An easier solution would be to use Docker.
...
+2. Stop it with `sudo systemctl stop redis`.
+If this doesn't work, you can check and manually kill the instance using port 6379 by running
+`ps aux | grep redis`
+
+```
+user     2570  0.0  0.0 253100  4908 pts/1    Sl+  13:11   0:00 redis-server *:6379
+user     2588  0.0  0.0 215744   888 pts/0    S+   13:16   0:00 grep --color=auto redis
+```
+Run `sudo kill -9 2570`
```
Comment (miggs125) [inline docs/CONTRIBUTING.md]: Again, this is unnecessary. See my comment above. FYI the process ID that you have added will likely not be the same for others running the same process. Suggesting an uninformed user to execute ```kill -9``` with ```sudo``` on a specific pid like you are suggesting here is dangerous and could cause some major issues on their machine.

**Comment by SeanPrashad** [comment]
Hey @cindyledev - there was some interesting work here and I'm sad to see that you closed your patch 😔I'd like to pair and continue improving the contributing docs, if you're up for it! Feel free to message me on slack - my username is `sprashad` 😀

**Pull request closed without merging**